objcgen/helpers/objcgenerator-helpers.cs

```diff
@@ -0,0 +1,52 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+using IKVM.Reflection;
+using Type = IKVM.Reflection.Type;
+
+using Embeddinator;
+namespace ObjC {
+	public partial class ObjCGenerator {
+
+		public static string GetObjCConstructorName (ParameterInfo [] parameters)
+		{
+			StringBuilder name = new StringBuilder ("init");
+			foreach (var p in parameters) {
+				if (name.Length == 4)
+					name.Append ("With");
+				else
+					name.Append (' ');
+				name.Append (PascalCase (p.Name));
+				name.Append (":(").Append (GetTypeName (p.ParameterType)).Append (") ").Append (p.Name);
+			}
+			return name.ToString ();
+		}
```

Comment:
so my PR got merged before yours ;-)

The code above can't be reused anymore. The new method would be a lot costlier to call (since it does the mono signature too). I think you should keep this one, rename it, make it private and simplify it.

IOW you only need a constant string for a signature so:
* it does not have to be pretty strings (no `init` or `With`);
* the parameter names are not needed (in fact they could differ between the types and still be the same .ctor);
* casing is not important;